libkbfs/repeated_wait_group.go

```diff
@@ -20,6 +20,12 @@ type RepeatedWaitGroup struct {
 	lock     sync.Mutex
 	num      int
 	isIdleCh chan struct{} // leave as nil when initializing
+	// TODO: we could remove this paused bool by converting the
```

Comment:
Ah, okay. I agree that we should try not to make `RepeatedWaitGroup` need to be initialized. How about this?

1) Have `pauseInitialized` and `onPauseCh`, which both start off as zero. The idea is that if `pauseInitialized` is `false`, then we pretend that `onPauseCh` is initialized, and once it is set to `true` (which happens the first time `Pause` or `WaitUnlessPaused` is called), then we maintain the invariant that `onPauseCh` is non-nil when we're not paused, and closed and set to nil when paused.
2) On Pause, if `pauseInitialized` is `false`, set it to `true` and return. Otherwise, close `onPauseCh` and set it to `nil`, or panic if it's already `nil`.
3) On Resume, if `pauseInitialized` is `false`, panic. Otherwise, re-initialize `onPauseCh` or panic if it's already re-initialized.
4) In `WaitUnlessPaused`, if `pauseInitialized` is `false`, set it to `true` and initialize `onPauseCh`. Then if `onPauseCh` is `nil`, we're paused so return immediately. Otherwise, wait on `onPauseCh` as part of the select.

I think this is pretty close to the current code. I'm just wary of having to keep `paused` and `pauseCh` in sync, and of remembering that `pauseCh` is nil when paused. What do you think?